packages/block-library/src/social-link/edit.js

```diff
@@ -128,19 +130,19 @@ const SocialLinkEdit = ( {
 				<PanelBody
 					title={ sprintf(
 						/* translators: %s: name of the social service. */
-						__( '%s label' ),
+						__( '%s link text' ),
 						socialLinkName
 					) }
 					initialOpen={ false }
 				>
 					<PanelRow>
 						<TextControl
 							__nextHasNoMarginBottom
-							label={ __( 'Link label' ) }
+							label={ __( 'Link text' ) }
 							help={ __(
-								'Briefly describe the link to help screen reader users.'
+								'The link text is only visible when Show links text is enabled.'
```

Comment:
Perhaps something a bit clearer, like "Visible when enabled from the parent Social Icons block."